tailnet/test/integration/integration.go

```diff
@@ -126,8 +156,10 @@ func (o SimpleServerOptions) Router(t *testing.T, logger slog.Logger) *chi.Mux {
 	})
 	require.NoError(t, err)
 
-	derpServer := derp.NewServer(key.NewNode(), tailnet.Logger(logger.Named("derp")))
-	derpHandler, derpCloseFunc := tailnet.WithWebsocketSupport(derpServer, derphttp.Handler(derpServer))
+	derpSrv := derp.NewServer(key.NewNode(), tailnet.Logger(logger.Named("derp")))
+	derpServer := atomic.Pointer[derp.Server]{}
+	derpServer.Store(derpSrv)
+	derpHandler, derpCloseFunc := tailnet.WithWebsocketSupport(derpServer.Load(), derphttp.Handler(derpServer.Load()))
```

Comment:
These should probably be stored in atomic pointers as well. You could make a `derpServer` wrapper struct that stores all 3 of these so you only need a single `atomic.Pointer`

Reply:
And then you could write a nice `newDerpServer` fn, as well as a `Close` method that calls the underlying close methods